test/utils/PinningUtils-test.ts

```diff
@@ -0,0 +1,253 @@
+/*
+ * Copyright 2024 The Matrix.org Foundation C.I.C.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { EventTimeline, EventType, IEvent, MatrixClient, MatrixEvent, Room } from "matrix-js-sdk/src/matrix";
+import { mocked } from "jest-mock";
+
+import { createTestClient } from "../test-utils";
+import PinningUtils from "../../src/utils/PinningUtils";
+import SettingsStore from "../../src/settings/SettingsStore";
+import { canPinEvent, isContentActionable } from "../../src/utils/EventUtils";
+import { ReadPinsEventId } from "../../src/components/views/right_panel/types";
+
+// The mock factory returns a mocked function
```

Comment:
I'm not sure what this comment is talking about?

Reply:
It should be deleted :)